### DISCUSSION TITLE
O aplicativo ReadKey InvalidOperationException não possui um console

### DISCUSSION THREAD

**Discussion by Davins-016** [balta-io/forum#471] [Tira Dúvidas] [answered]
Olá
Eu estou fazendo o curso de introdução ao C# da plataforma. Enquanto eu fazia o  Módulo "Mão na Massa: Editor de textos", eu me deparei com uma problema durante a confecção do programa.
O problema supracitado é este :  "Exception has occurred: CLR/System.InvalidOperationException
An unhandled exception of type 'System.InvalidOperationException' occurred in System.Console.dll: 'Cannot read keys when either application does not have a console or when console input has been redirected. Try Console.Read.'
   at System.ConsolePal.ReadKey(Boolean intercept)"
   Não sei o porque deste problema, pois o meu código é condizente com o apresentado durante o módulo. Preciso de ajuda para solucionar este problema.

Código feito por mim:

    private static void Main(string[] args){
    {
        Menu();

    }
    static void Menu(){
        Console.WriteLine("O que você deseja fazer?");
        Console.WriteLine("1- Abrir arquivo");
        Console.WriteLine("2- Criar novo arquivo");
        Console.WriteLine("0 - Sair");
        short opcao = short.Parse(Console.ReadLine());
        switch(opcao){
            case 0: System.Environment.Exit(0);break;
            case 1: Abrir();break;
            case 2: Editar();break;
            default: Menu();break;
        }
        
        }

    }
    static void Abrir(){
        Console.WriteLine("Qual caminho do arquivo?");
        string caminho = Console.ReadLine();

        using(var arquivo = new StreamReader(caminho)){
        String texto = arquivo.ReadToEnd();
        Console.WriteLine(texto);
        }
        Console.WriteLine(" ");
        Console.ReadLine();
        Menu();
    }

    static void Editar(){

        Console.WriteLine("Digite seu texto aqui em baixo (ESC para sair)");
        Console.WriteLine("---------------------");
        string texto = "";

        do{
            texto += Console.ReadLine();
            texto += Environment.NewLine;
        }
        while(Console.ReadKey().Key != ConsoleKey.Escape);
        Salvar(texto);
    }

    static void Salvar(string texto)
    {

        Console.WriteLine("Qual caminho para salvar o arquivo?");
        var path = Console.ReadLine();

        using(var arquivo = new StreamWriter(path))
        {
           arquivo.Write(texto);
        }

     Console.WriteLine($"Arquivo {path} salvo com sucesso!");
     Console.ReadLine();
        Menu();
        
    }

    private static void Menu() => throw new NotImplementedException();

    private string GetDebuggerDisplay()
    {
        return ToString();
    }
}

Agradeço desde já pela assistência.

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @Davins-016, como vai? Em qual linha do seu código este problema aconteceu?

**Reply by Davins-016** [reply]
Bom dia também, estou bem e você? O problema acontece nesta linha while(Console.ReadKey().Key != ConsoleKey.Escape);

**Reply by andrebaltieri** [reply]
Bom dia, @Davins-016, como vai? Você está debugando a aplicação pelo Visual Studio Code? Se sim, tenta para a depuração e iniciar o app pelo comando `dotnet run` como mostrado no curso.

**Reply by Davins-016** [reply]
Bom dia @andrebaltieri, estou bem e você. O problema foi resolvido com a sua recomendação, obrigado!
